### ISSUE TITLE
[Feature Anywhere] Support fine-grained error states from `VisLayer`s

### ISSUE THREAD

**Issue by ohltyler** [opensearch-project/OpenSearch-Dashboards#3267]
We need to support errors at a per-plugin-resource (or per `VisLayer`) granularity. This way, this can be propagated when rendering the vis to show some error states and/or within the plugin flyouts, such as showing an error state for a particular detector in a detector list, or error icons next to a detector in the View Events page.

**Comment by ohltyler** [comment]
A proposed solution is that we have 2 different error types:
1. Security-related errors from fetching a plugin resource and/or results. 
2. General errors out of the user's control, such as request timeout, issues connecting to the cluster, etc.
For security errors, because any combination will prevent regular rendering on the view events flyout (need name from get-resource api call, and results from get-results api call), we will hide that VisLayer and it won't be one of the options in the View Events flyout.

For general errors, we can still show the plugin resource details, and have an error icon that can provide details on why plugin details and/or results couldn't be fetched. This can at least provide some transparency to the user. Since we have a 'refresh' button in the flyout, this gives the user the option to click that and potentially successfully load data that had errors during the initial render of the flyout.

**Comment by ohltyler** [comment]
Examples (`message` field may be omitted or not needed)
```
{
   type: ERROR_TYPE.SECURITY,
   pluginResourceID: abc123
   message?: "User x has no permissions to read detector abc123"
}
```
```
{
   type: ERROR_TYPE.GENERAL,
   pluginResourceID: abc123
   message?: "Failed to fetch data for detector abc123"
}
```

**Comment by ohltyler** [comment]
Closed via #3646
